pkg/rancher-desktop/utils/dockerUtils.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * The return result of parseImageReference().
+ *
+ * @note This is only exported for the test.
+ */
+export class imageInfo {
+  /**
+   * The registry, as a URL (e.g. `https://registry.opensuse.org/`);
+   * defaults to Docker Hub, i.e. `https://index.docker.io/`.
+   */
+  registry: URL;
+  /**
+   * The image name (e.g. `opensuse/leap`).
+   * For Docker Hub images, `library/` will be added if there is no org.
+   */
+  name: string;
+  /** Any tags (e.g. `latest`, `15.4`) */
+  tag?: string;
+
+  constructor(registry: URL, name: string, tag?: string) {
+    this.registry = registry;
+    this.name = name;
+    this.tag = tag;
+  }
+
+  /**
+   * Check if this image (excluding the tag) is the same as another one.
+   */
+  equalName(other?: imageInfo | null): boolean {
+    return this.registry.href === other?.registry.href && this.name === other?.name;
+  }
+}
+
+/**
+ * makeRE is a tagged template for making regular expressions with /x (i.e.
+ * ignoring any whitespace within the regular expression itself).
+ * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates
+ */
+function makeRE(strings: TemplateStringsArray, ...substitutions: any[]) {
+  const substitutionSources = substitutions.map(s => s instanceof RegExp ? s.source : s);
+  const raw = String.raw(strings, ...substitutionSources);
+  const lines = raw.split(/\r?\n/);
+  // Drop comments at end of line
+  const uncommentedLines = lines.map(line => line.replace(/\s#.*$/, ''));
+
+  return new RegExp(uncommentedLines.join('').replace(/\s+/g, ''));
+}
+
+/**
+ * ImageNameRegExp is a regular expression that matches a docker image name
+ * (including optional registry and one or more name components).
+ */
+const ImageNameRegExp = (function() {
+  const domainComponent = makeRE`
+    # a domain component is alpha-numeric-or-dash, but the start and end
```

Comment:
These in-pattern comments should be moved before the `makeRE` part and changed to `// ...` comments.
That should be easier to read in IDEs because they have no particular syntax in a backquoted string (unlike `${...}` things).

Reply:
Moved the comments out.
`domain` still needs to use `makeRE` because it pulls in `domainComponent`.

Dropped `nameRE` (returning it directly).